packages/mongodb-chatbot-eval/src/withConfig.ts

```diff
@@ -0,0 +1,162 @@
+import yargs from "yargs";
+import Path from "path";
+import { logger } from "mongodb-rag-core";
+import { EvalConfig } from "./EvalConfig";
+
+export type LoadConfigArgs = {
+  config?: string;
+};
+
+export const loadConfig = async ({
+  config: configPathIn,
+}: LoadConfigArgs): Promise<EvalConfig> => {
+  const path = Path.resolve(
+    configPathIn === undefined ? "eval.config.cjs" : configPathIn
+  );
+
+  const partialConfigConstructor = (await import(path))
+    .default as () => Promise<Partial<EvalConfig>>;
+  const partialConfig = await partialConfigConstructor();
+
+  const missingProperties: string[] = [];
+  const config: EvalConfig = {
+    ...partialConfig,
+    metadataStore: checkRequiredProperty(
+      partialConfig,
+      "metadataStore",
+      missingProperties
+    ),
+    generatedDataStore: checkRequiredProperty(
+      partialConfig,
+      "generatedDataStore",
+      missingProperties
+    ),
+    evaluationStore: checkRequiredProperty(
+      partialConfig,
+      "evaluationStore",
+      missingProperties
+    ),
+    reportStore: checkRequiredProperty(
+      partialConfig,
+      "reportStore",
+      missingProperties
+    ),
+    commands: checkRequiredProperty(
+      partialConfig,
+      "commands",
+      missingProperties
+    ),
+  };
+
+  if (missingProperties.length !== 0) {
+    throw new Error(
+      `Config is missing the following properties: ${missingProperties.join(
+        ", "
+      )}`
+    );
+  }
+
+  return config;
+};
+
+export const withConfig = async <T>(
+  action: (config: ResolvedConfig, args: T) => Promise<void>,
+  args: LoadConfigArgs & T
+) => {
+  const config = await loadConfig(args);
+  const [resolvedConfig, cleanup] = await resolveConfig(config);
+  try {
+    return await action(resolvedConfig, args);
+  } finally {
+    await Promise.all(
+      cleanup.map(async (close) => {
+        try {
+          await close();
+        } catch (error) {
+          logger.error(`Cleanup failed: ${(error as Error).message}`);
+        }
+      })
+    );
+  }
+};
+
+/**
+  Apply config options to CLI command.
+ */
+export const withConfigOptions = <T>(
+  args: yargs.Argv<T>
+): yargs.Argv<T & LoadConfigArgs> => {
+  return args.option("config", {
+    string: true,
+    description: "Path to config JS file.",
+  });
+};
+
```

Comment:
```suggestion
// NOTE: consider moving the generic configurable CLI stuff to `mongodb-rag-core` or another independent library since we're using across multiple projects.
```